core/src/main/java/org/apache/iceberg/actions/BaseRewriteDataFilesAction.java

```diff
@@ -271,11 +273,13 @@ public RewriteDataFilesActionResult execute() {
   private void replaceDataFiles(Iterable<DataFile> deletedDataFiles, Iterable<DataFile> addedDataFiles,
                                 long startingSnapshotId) {
     try {
-      RewriteFiles rewriteFiles = table.newRewrite()
-          .validateFromSnapshot(startingSnapshotId)
-          .rewriteFiles(Sets.newHashSet(deletedDataFiles), Sets.newHashSet(addedDataFiles));
-      commit(rewriteFiles);
+      doReplace(deletedDataFiles, addedDataFiles, startingSnapshotId);
+    } catch (CommitStateUnknownException e) {
+      LOG.warn("Commit state unknown for files: {}, cannot clean up files because they may have been committed " +
+          "successfully.", Lists.newArrayList(addedDataFiles), e);
+      throw e;
     } catch (Exception e) {
+      LOG.warn("Failed to rewrite, files {} will be delete", Lists.newArrayList(addedDataFiles));
```

Comment:
Let's remove the file list here as well. Instead, use "Failed to commit rewrite, cleaning up rewritten files"